app/code/Bulbulatory/Recommendations/Model/Total/Quote/RecommendationDiscount.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Bulbulatory\Recommendations\Model\Total\Quote;
+
+use Bulbulatory\Recommendations\Helper\Discount;
+use Magento\Framework\Pricing\PriceCurrencyInterface;
+use Magento\Quote\Api\Data\ShippingAssignmentInterface;
+use Magento\Quote\Model\Quote;
+use Magento\Quote\Model\Quote\Address\Total;
+use Magento\Quote\Model\Quote\Address\Total\AbstractTotal;
+use Bulbulatory\Recommendations\Helper\Config;
+
+/**
+ * Class RecommendationDiscount
+ * @package Bulbulatory\Recommendations\Model\Total\Quote
+ */
+class RecommendationDiscount extends AbstractTotal
+{
+    /**
+     * @var PriceCurrencyInterface
+     */
+    protected $priceCurrency;
+
+    /**
+     * @var bool
+     */
+    private $isModuleEnabled = false;
+
+    /**
+     * @var int
+     */
+    private $discount = 0;
+
+    /**
+     * RecommendationDiscount constructor.
+     * @param PriceCurrencyInterface $priceCurrency
+     * @param Config $configHelper
+     * @param Discount $discountHelper
+     */
+    public function __construct(
+        PriceCurrencyInterface $priceCurrency,
+        Config $configHelper,
+        Discount $discountHelper
+    ){
+        $this->priceCurrency = $priceCurrency;
+
+        if($configHelper->isModuleEnabled())
+        {
+            $this->isModuleEnabled = true;
+
+            $discountPercent = $discountHelper->getRecommendationDiscountPercent();
+            $this->discount = $discountPercent/100;
+        }
+    }
+
+    /**
+     * @param Quote $quote
+     * @param ShippingAssignmentInterface $shippingAssignment
+     * @param Total $total
+     * @return $this|bool
+     */
+    public function collect(
+        Quote $quote,
+        ShippingAssignmentInterface $shippingAssignment,
+        Total $total
+    )
+    {
+        parent::collect($quote, $shippingAssignment, $total);
+
+        if($this->isModuleEnabled)
+        {
+            $baseDiscount = $this->discount * $total->getSubtotal();
+            $discount =  $this->priceCurrency->convert($baseDiscount);
+            $total->addTotalAmount($this->getCode(), -$discount);
+            $total->addBaseTotalAmount($this->getCode(), -$baseDiscount);
+            $total->setBaseGrandTotal($total->getBaseGrandTotal() - $baseDiscount);
+            $quote->setCustomDiscount(-$discount);

```

Comment:
If total name is eg. `my_super_custom_discount` then the method setting it should be named `->setMySuperCustomDiscount(..)`. When you change your total name, remember to also give this method a proper name.

Reply:
@TM18

> If total name is eg. my_super_custom_discount then the method setting it should be named ->setMySuperCustomDiscount(..). When you change your total name, remember to also give this method a proper name.

I changed _setCustomDiscount_ to _setCustomRecommendationDiscount_